### PR TITLE
Deprecate not providing mod.conf

### PR DESCRIPTION
**Pull request by rubenwardy**
This adds a deprecation message when:

* Mods don't provide a mod.conf with the name set
* Mods set dependencies in depends.txt but not in mod.conf
* Mods set description in description.txt but not in mod.conf

## How to test

Load some mods

### CODE REVIEW

File: src/content/mods.cpp [hunk 1]
```diff
@@ -109,6 +128,10 @@ void parseModContents(ModSpec &spec)
 			std::vector<std::string> dependencies;
 
 			std::ifstream is((spec.path + DIR_DELIM + "depends.txt").c_str());
+
+			if (is.good())
+				log_mod_deprecation(spec, "depends.txt is deprecated, please use mod.conf instead.");
```
Comment (v-rob) [inline src/content/mods.cpp:133]: If `mod.conf` exists and has a `depends`/`optional_depends` key, and `depends.txt` also exists, no warning will be produced.  Same goes for `description.txt`.  Is this intended?

**Review by v-rob — COMMENTED** [review]
Looks good and works as expected, just one question.

File: src/content/mods.cpp (hunk 1 above)
Comment (v-rob) [reply src/content/mods.cpp:133]: Oh, never mind, I see.  This is to allow mods to work on older versions of Minetest where the txt files were necessary without deprecation notices on newer versions.

**Comment by sfan5** [comment]
Is this time-sensitive enough that it should go into 5.4?

**Comment by rubenwardy** [comment]
It doesn't need to, I have no plans to remove support for depends.txt and mod.conf. Up to you

**Comment by Desour** [comment]
Shouldn't such warnings only be shown for mods that are activated for the world?

I get 720 lines on stderr just by starting a singleplayer game and closing immediately afterwards. And I don't want to maintain all the old mods that I have lying around.

**Comment by rubenwardy** [comment]
That's a good point, I'll make an issue for it

**Pull request merged**